src/app/services/actions/autocomplete-action-creators.ts

```diff
@@ -30,62 +31,28 @@ export function fetchAutoCompleteOptions(url: string): Function {
       'Content-Type': 'application/json',
     };
 
-    let permissionsUrl = `${devxApi}/openapi`;
-    permissionsUrl = `${permissionsUrl}?url=/${url}&style=geautocomplete&format=json`;
+    let openApiUrl = `${devxApi}/openapi`;
+    openApiUrl = `${openApiUrl}?url=/${url}&style=geautocomplete&format=json`;
 
     const options: IRequestOptions = { headers };
 
     dispatch(fetchAutocompletePending());
 
     try {
-      const response = await fetch(permissionsUrl, options);
+      const response = await fetch(openApiUrl, options);
       if (response.ok) {
         const autoCompleteOptions = await response.json();
-        const params = {
+        const parameters = {
           options: autoCompleteOptions,
           url,
           verb: sampleQuery.selectedVerb.toLowerCase()
         };
-        const reduced = getReducedVersion(params);
-
-        return dispatch(fetchAutocompleteSuccess(reduced));
+        const parsedResponse = parseOpenApiResponse(parameters);
```

Comment:
This can be an error object instead of `parsedResponse`. `parseOpenApiResponse` returns an error when it occurs. Did you mean to throw the error in `parseOpenApiResponse`?

Reply:
I should have actually... nice catch